common/lib/dependabot/metadata_finders/base/release_finder.rb

```diff
@@ -252,36 +252,38 @@ def previous_version
           end
 
           if dependency.previous_version.match?(/^[0-9a-f]{40}$/) &&
-             ref_changed?
+             ref_changed? && previous_ref
```

Comment:
We could have gone to `nil` `previous_ref` here so we should return the `previous_version` instead